ledger/src/blockstore.rs

```diff
@@ -2055,7 +2055,7 @@ impl Blockstore {
         );
         let confirmed_unrooted_slots: Vec<_> =
             AncestorIterator::new_inclusive(highest_confirmed_slot, self)
-                .filter(|&slot| slot > self.last_root())
+                .take_while(|&slot| slot > self.last_root())
```

Comment:
hoisting `self.last_root()` outside the loop might be better even

Reply:
Good call, will follow with that change; thanks!